backend/lib/rpc-service/service.js

```diff
@@ -0,0 +1,91 @@
+import { RPC_URL } from '../config.js'
+import { base64pad } from 'multiformats/bases/base64'
+import { encode as cborEncode } from '@ipld/dag-cbor'
+import { decode as jsonDecode } from '@ipld/dag-json'
+import { request } from 'undici'
+import { rawEventEntriesToEvent } from './utils.js'
+import { Value } from '@sinclair/typebox/value'
+import { ClaimEvent, RawActorEvent, BlockEvent } from './data-types.js'
+
+/** @import {CID} from 'multiformats' */
+
+/**
+ * @param {string} method
+ * @param {Object} params
+  * @returns {Promise<object>}
+  */
+export const rpcRequestFn = async (method, params) => {
+  const reqBody = JSON.stringify({ method, params, id: 1, jsonrpc: '2.0' })
+  const response = await request(RPC_URL, {
+    bodyTimeout: 1000 * 60,
+    headersTimeout: 1000 * 60,
+    method: 'POST',
+    headers: { 'content-type': 'application/json' },
+    body: reqBody
+  })
+  return jsonDecode(new Uint8Array(await response.body.arrayBuffer())).result
+}
+/**
+     * @param {object} actorEventFilter
+     * Returns actor events filtered by the given actorEventFilter
+     * @returns {Promise<Array<BlockEvent>>}
+     */
+export async function getActorEvents (actorEventFilter, makeRpcRequest) {
+  const rawEvents = (await makeRpcRequest('Filecoin.GetActorEventsRaw', [actorEventFilter]))
+  if (!rawEvents || rawEvents.length === 0) {
+    console.log(`No actor events found in the height range ${actorEventFilter.fromHeight} - ${actorEventFilter.toHeight}.`)
+    return []
+  }
+  // TODO: handle reverted events
```

Comment:
Shall we open an issue for this TODO, and add it to the current or the next milestone?

Reply:
I created [one](https://github.com/filecoin-station/deal-observer/issues/22).

Reply:
Nitpick:

We usually mention the issue in the code comment.

```suggestion
  // TODO: handle reverted events
  // https://github.com/filecoin-station/deal-observer/issues/22
```